### PR TITLE
SONARIAC-1131 Report secondary locations in values.yaml for existing Kubernetes checks

### PR DESCRIPTION
**Pull request by petertrr**

### CODE REVIEW

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/helm/tree/utils/GoTemplateAstHelper.java [hunk 1]
```diff
@@ -46,6 +46,7 @@ public static List<ValuePath> findNodes(GoTemplateTree tree, TextRange range, St
 
     return allChildren(nodes).stream()
       .filter(FieldNode.class::isInstance)
+      .filter(hasOverlayingLocation(location))
```
[the remark this answers is not in the document]
Comment (petertrr) [reply iac-extensions/kubernetes/src/main/java/org/sonar/iac/helm/tree/utils/GoTemplateAstHelper.java:49]: Yes, that was exactly the problem. On line 44 we get a top-level overlapping node, that can be quite big. I faced a problem in the file which was one big `{{ if ... }}` for 500 lines. The old logic was returning 200 `.Value` nodes here, even though only one was really overlapping the issue site.

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/SecondaryLocationLocator.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+public class SecondaryLocationLocator {
+  private static final Logger LOG = LoggerFactory.getLogger(SecondaryLocationLocator.class);
+  private final YamlParser yamlParser;
+
+  public SecondaryLocationLocator(YamlParser yamlParser) {
+    this.yamlParser = yamlParser;
+  }
+
+  public List<SecondaryLocation> maybeFindSecondaryLocationsInAdditionalFiles(InputFileContext inputFileContext, TextRange shiftedTextRange) {
```
[the remark this answers is not in the document]
Comment (petertrr) [reply iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/SecondaryLocationLocator.java:50]: Yes, good catch, after extracting this class it makes sense to rename the entrypoint method.

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/SecondaryLocationLocatorTest.java [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+  @RegisterExtension
+  public LogTesterJUnit5 logTester = new LogTesterJUnit5().setLevel(Level.DEBUG);
+
+  private final SecondaryLocationLocator secondaryLocationLocator = new SecondaryLocationLocator(new YamlParser());
+
+  @Test
+  void shouldFindSecondaryLocationsInValuesFile() {
+    var inputFileContext = inputFileContextWithTree();
+
+    var locationsInAdditionalFiles = secondaryLocationLocator.findSecondaryLocationsInAdditionalFiles(inputFileContext, TextRanges.range(1, 6, 1, 22));
```
Comment (mstachniuk) [inline iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/SecondaryLocationLocatorTest.java:69]: static import

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/SecondaryLocationLocatorTest.java [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+    Assertions.assertThat(locationsInAdditionalFiles).hasSize(1);
```
Comment (mstachniuk) [inline iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/SecondaryLocationLocatorTest.java:71]: static import

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/SecondaryLocationLocatorTest.java [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+    var actionNode = new ActionNodeImpl(8, 15, pipeNode);
+    var textNode = new TextNodeImpl(0, 5, "bar: ");
+    ListNodeImpl root = new ListNodeImpl(0, 15, List.of(textNode, actionNode));
+    var goTemplateTree = new GoTemplateTreeImpl("test", "test", 0, root);
+    inputFileContext.setGoTemplateTree(goTemplateTree);
+
+    return inputFileContext;
+  }
+
+  @CheckForNull
```
Comment (mstachniuk) [inline iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/SecondaryLocationLocatorTest.java:201]: such annotation is not needed in tests I guess

**Review by mstachniuk — APPROVED** [review]
LGTM, don't forget to commit the improvements

**Pull request merged**